File: dataset_reduzido/avatar/Python/Code/codeforces_55_B.py
def rememberTheNumber ( nums , i , operations , ans ) :
    if i >= 3 :
        return
    for j in range ( 4 ) :
        for k in range ( j + 1 , 4 ) :
            if ( nums [ j ] != - 1 and nums [ k ] != - 1 ) :
                s = nums [ j ]
                nums [ j ] = - 1
                t = nums [ k ]
                if ( operations [ i ] == '+' ) :
                    nums [ k ] = s + t
                elif ( operations [ i ] == '*' ) :
                    nums [ k ] = s * t
                if ( i == 2 and nums [ k ] < ans [ 0 ] ) :
                    ans [ 0 ] = nums [ k ]
                rememberTheNumber ( nums , i + 1 , operations , ans )
                nums [ j ] = s
                nums [ k ] = t
nums = list ( map ( int , input ( ).split ( ) ) )
operations = list ( map ( str , input ( ).split ( ) ) )
ans = [ 10000000000000000000000000 ]
rememberTheNumber ( nums , 0 , operations , ans )
print ( ans [ 0 ] )

File: output/Qwen_repaired_1/avatar/Python/Java/codeforces_55_B.java
import java.util.Scanner;

public class codeforces_55_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int[] nums = Arrays.stream(scanner.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        String[] operations = Arrays.stream(scanner.nextLine().split(" ")).toArray();
        long[] ans = new long[]{Long.MAX_VALUE};
        rememberTheNumber(nums, 0, operations, ans);
        System.out.println(ans[0]);
    }

    public static void rememberTheNumber(int[] nums, int i, String[] operations, long[] ans) {
        if (i >= operations.length) {
            return;
        }
        for (int j = 0; j < nums.length; j++) {
            for (int k = j + 1; k < nums.length; k++) {
                if (nums[j] != -1 && nums[k] != -1) {
                    int tempJ = nums[j], tempK = nums[k];
                    nums[j] = -1;
                    switch (operations[i]) {
                        case "+":
                            nums[k] = tempJ + tempK;
                            break;
                        case "*":
                            nums[k] = tempJ * tempK;
                            break;
                    }
                    if (i == operations.length - 1 && Math.abs((long) nums[k]) < ans[0]) {
                        ans[0] = Math.abs((long) nums[k]);
                    }
                    rememberTheNumber(nums, i + 1, operations, ans);
                    nums[j] = tempJ;
                    nums[k] = tempK;
                }
            }
        }
    }
}